sdks/java/core/src/main/java/org/apache/beam/sdk/util/MoreFutures.java

```diff
@@ -0,0 +1,233 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.util;
+
+import com.google.auto.value.AutoValue;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ForkJoinPool;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+
+/**
+ * Utilities to do future programming with Java 8.
+ *
+ * <p>Standards for these utilities:
+ *
+ * <ul>
+ *   <li>Always allow thrown exceptions, and they should cause futures to complete exceptionally.
+ *   <li>Always return {@link CompletionStage} as a future value.
+ *   <li>Return {@link CompletableFuture} only to the <i>producer</i> of a future value.
+ * </ul>
+ */
+public class MoreFutures {
+
+  /** Like {@link Runnable} but allowed to throw any exception. */
```

Comment:
Clarify: are these interfaces intended to be used only in futures-related code? (if they are intended to be used more widely, they should be outside `MoreFutures`; if not, ideally that should be documented, cause it's tempting)

Reply:
In fact, these exist within the SDK harness, among other places, already, and I don't believe all of those uses interact directly with futures; I'd rather just have them in `util`, or `function`, (or any other meaningful name)

Reply:
Done